include/oneapi/dpl/pstl/pstl_config.h

```diff
@@ -105,13 +106,27 @@
 #define _PSTL_EARLYEXIT_PRESENT (__INTEL_COMPILER >= 1800)
 #define _PSTL_MONOTONIC_PRESENT (__INTEL_COMPILER >= 1800)
 
-#if (__INTEL_COMPILER >= 1900 || !defined(__INTEL_COMPILER) && _PSTL_GCC_VERSION >= 40900 || _OPENMP >= 201307)
+#if (__INTEL_LLVM_COMPILER || __INTEL_COMPILER >= 1900 ||                                                              \
+     !defined(__INTEL_LLVM_COMPILER) && !defined(__INTEL_COMPILER) && _ONEDPL_GCC_VERSION >= 40900 ||                  \
+     _OPENMP >= 201307)
 #    define _PSTL_UDR_PRESENT 1
 #else
 #    define _PSTL_UDR_PRESENT 0
 #endif
 
-#define _PSTL_UDS_PRESENT (__INTEL_COMPILER >= 1900 && __INTEL_COMPILER_BUILD_DATE >= 20180626)
+// UDS is broken with __INTEL_LLVM_COMPILER on Windows,
+// however it may be fixed in the future releases (ones which follow 20240000)
+#if (defined(_MSC_VER) && __INTEL_LLVM_COMPILER > 20240000) || (!defined(_MSC_VER) && __INTEL_LLVM_COMPILER > 20230100)
+#    define _ONEDPL_INTEL_LLVM_UDS_PRESENT 1
```

Comment:
Why do we introduce `_ONEDPL_*` macro in `pstl_config.h` ?
the upstream pstl version has just `_PSTL_*` macro everywhere.

Reply:
Thanks for pointing it out. I have replaced ONDPL with PSTL in pstl_config.